ttf2mesh.c

```diff
@@ -1314,7 +1325,7 @@ static int locate_fmt4_table(pps_t *s)
     for (i = 0; i < ntab; i++)
     {
         int offset = big32toh(s->pcmap->encRecs[i].offset);
-        if (offset + 4 > s->scmap) return TTF_ERR_FMT;
+        if (offset + 4 > s->scmap || offset < 0) return TTF_ERR_FMT;
```

Comment:
Not 100% sure about this, but I guessed offset be always non-negative? Some inputs caused it to be <0

Reply:
You're right. The following check is probably the best way to check for overflow:
`if (offset + 4 > s-> scmap || offset + 4 < 0) return TTF_ERR_FMT;`